conjure-go-client/httpclient/metrics.go

```diff
@@ -56,18 +49,32 @@ const (
 var (
 	MetricTagConnectionNew    = metrics.MustNewTag("reused", "false")
 	MetricTagConnectionReused = metrics.MustNewTag("reused", "true")
+
+	metricTagFamily1xx     = metrics.MustNewTag(metricTagFamily, "1xx")
+	metricTagFamily2xx     = metrics.MustNewTag(metricTagFamily, "2xx")
+	metricTagFamily3xx     = metrics.MustNewTag(metricTagFamily, "3xx")
+	metricTagFamily4xx     = metrics.MustNewTag(metricTagFamily, "4xx")
+	metricTagFamily5xx     = metrics.MustNewTag(metricTagFamily, "5xx")
+	metricTagFamilyOther   = metrics.MustNewTag(metricTagFamily, "other")
+	metricTagFamilyTimeout = metrics.MustNewTag(metricTagFamily, "timeout")
 )
 
 // A TagsProvider returns metrics tags based on an http round trip.
 type TagsProvider interface {
-	Tags(*http.Request, *http.Response) metrics.Tags
+	Tags(*http.Request, *http.Response, error) metrics.Tags
 }
 
 // TagsProviderFunc is a convenience type that implements TagsProvider.
-type TagsProviderFunc func(*http.Request, *http.Response) metrics.Tags
+type TagsProviderFunc func(*http.Request, *http.Response, error) metrics.Tags
```

Comment:
Mind adding a note as to what this error represents? It's not clear at first that this is intended to be the response error after a round trip